### DISCUSSION TITLE
spdif esp32

### DISCUSSION THREAD

**Discussion by parabyte** [pschatzmann/arduino-audio-tools#1501] [Q&A] [answered]
Is there any way to get spdif data into the esp32 with just the optical receiver i have in my junk box, without a decoding ic?

**Comment by pschatzmann** [comment, marked as answer]
I guess the challenge will be to decode the Manchester encoded data provided by spdif fast enough.
After that you would need to decode the S/PDIF protocol

**Comment by parabyte** [comment]
i been reading through a thread about one of the stm32's that someone managed to do just that on.

but i see where your coming from, the overhead of your libs doing there fantastical things, and also on top of that trying to bitbang i2c or what ever the equivilent of rx of bitbangig is.

thanks for getting back to me.
